README.md

```diff
@@ -17,76 +16,126 @@ _A table of contents to help people find things_
+
+The only required options are listed in the example _above_. These are:
+- `type`: String. The o-message variant. The available variants are 'alert', 'alert-bleed' and 'alert-inner'.
+- `status`: String. Alert variants require a status, and the options are 'success', 'neutral' and 'error'.
+- `content`: Object. Holds the following values for text properties:
+	- `highlight`: String. The highlighted text in a message. Defaults to `null`
+	-	`detail`: String. The detail about the nature of a message.
+
+The following options are not required, and all have a default value:
+
+- `autoOpen`: Boolean. Whether to open the message automatically, defaults to `true`.
+- `messageClass`: String. The base class name for the component's elements, defaults to `o-message`.
+- `variant`: String. The message variant which allows a message to be a bleed message or an inline message. Defaults to `null`.
+- `parentElement`: String. This determines the element that the message will be appended to. If none is declared, it will automatically append to the body, or an element with the data attribute `data-o-component=o-message`, defaults to `null`.
+- `content`: Object. Holds the following values for text properties:
+	-	`additionalInfo`: String. More information about the message –  only applies to an `alert-inner` message. Defaults to `null`
+- `button`: Object. Holds the following values for button properties:
+	- `text`: String. text value of the button.
+	- `url`: String. The URL the button links to.
+- `link`: Object. Holds the following values for link properties:
+	- `text`: String. text value of the link.
+	- `url`: String. The URL the link links to.
+- `close`: Boolean. Whether or not to display the close button. Defaults to `true` for regular messages, to `false` for inline messages.
 
-## Contributing
-If your component is particularly complicated (image sets fall into this category) then a contributing section or even a contributing.md might be useful.
 
+### Sass
+As with all Origami components, o-message has a [silent mode](http://origami.ft.com/docs/syntax/scss/#silent-styles). To use its compiled CSS (rather than using its mixins with your own Sass) set `$o-message-is-silent: false;` in your Sass before you import the o-message Sass.
 
-## Migration guide
-_Migration guides are very important! Always include one for major releases. To create a codeblock that has diff highligting, use three backticks followed by the word diff_
-
-### Migrating from 1.X.X to 2.X.X
-
-The 2.0.0 release changes the default behaviour of o-component-boilerplate.
+o-message includes mixins that you can use if you'd rather _not_ have origami classnames in your page. These are only available if you're not using the Build Service:
 
-```diff
-<div class="o-component-boilerplate__container">
-- remove this line
-+ add this line
-</div>
+```sass
+@include oMessage($class: 'my-banner', $variant: 'alert-inner', $status: 'success');
```

Comment:
This could be `type` rather than `variant` consistent with the JS. Viewing a variant through a branded lens both the type and the status combined constitute the variant of the message.